src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Range.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class LocalRewriter
+    {
+        public override BoundNode VisitRangeExpression(BoundRangeExpression node)
+        {
+            Debug.Assert(node != null);
+
+            if (node.LeftOperand is null)
+            {
+                if (node.RightOperand is null)
+                {
+                    // Case 1:              ".."
+                    // translates to:       Range.All()
+
+                    MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__All);
+                    return MakeCall(node.Syntax, rewrittenReceiver: null, method, ImmutableArray<BoundExpression>.Empty, node.Type);
+                }
+
+                BoundExpression rightOperand = VisitExpression(node.RightOperand);
+                rightOperand = NullableAlwaysHasValue(rightOperand) ?? rightOperand;
+
+                if (node.RightOperand.Type.IsNullableType())
+                {
+                    // Case 2:              "..nullableRight"
+                    // translates to:       nullableRight.HasValue ? new Nullable(Range.ToEnd(nullableRight.GetValueOrDefault())) : default
+
+                    MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__ToEnd);
+                    return LiftRangeExpression(node.Syntax, node.Type, method, rightOperand);
+                }
+                else
+                {
+                    // Case 3:              "..right"
+                    // translates to:       Range.ToEnd(right)
+
+                    MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__ToEnd);
```

Comment:
>MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__ToEnd); [](start = 20, length = 109)

It feels like this can be moved out of the ```if``` and ```else```.

Reply:
I'd prefer to keep it for readability reasons.

---
In reply to: [185365559](https://github.com/dotnet/roslyn/pull/26405#discussion_r185365559) [](ancestors = 185365559)